<?xml version="1.0" encoding="UTF-8"?>

<launch>

    <node pkg="imu_calib" type="apply_calib" name="apply_calib" output="screen" respawn="false">
        <param name="calib_file" value="$(find robotin_project)/config/imu_calib.yaml" />
		<param name="calibrate_gyros" value="true" />
		<remap from="raw" to="raw_imu"/>
    </node>

    <!-- Filter and fuse raw imu data -->
    <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_madgwick" output="screen" respawn="false" >
	<param name="fixed_frame" value="base_link" />
	<param name="gain" value="0.1"/>
	<param name="zeta" value="0.0"/>
	<param name="mag_bias_z" value="0.0"/>	    
	 <param name="stateless" value="false"/> 
        <param name="use_mag" value="true" />
        <param name="publish_tf" value="false" />
        <param name="use_magnetic_field_msg" value="true" /> 
        <param name="world_frame" value="enu" />
        <param name="orientation_stddev" value="0.05"/>
    </node>

    <!-- Publish static transform from base_footprint to imu_link -->
    <node pkg="tf" type="static_transform_publisher" name="base_to_imu_broadcaster" args="0.072 -0.0205 0.122 0 0 0.0 base_link imu_link 100" />

</launch>
